From 6fc077746b7ca2728c2d3a52dcc0377198246c01 Mon Sep 17 00:00:00 2001
From: Yangbo Lu <yangbo.lu@freescale.com>
Date: Mon, 10 Aug 2015 15:09:22 +0800
Subject: [PATCH 439/451] mmc: sdhci-pltfm: remove
 SDHCI_QUIRK_BROKEN_CARD_DETECTION for ls1043a

Use interrupt mode to check card instead of polling mode.
This can reduce CMD interrupts occurred by polling, improve r/w
performance and avoid call trace caused by polling card status.

Signed-off-by: Yangbo Lu <yangbo.lu@freescale.com>
[Xulin: Original patch taken from
Linux-LS1043A-SDK-V0.4-SOURCE-20150826-yocto.iso]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi |    2 +-
 drivers/mmc/host/sdhci-of-esdhc.c              |    3 ++-
 drivers/mmc/host/sdhci-pltfm.c                 |    3 ++-
 3 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
index 89f212e..e72e203 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
@@ -132,7 +132,7 @@
 	};
 
 	esdhc: esdhc@1560000 {
-		compatible = "fsl,ls1021a-esdhc", "fsl,esdhc";
+		compatible = "fsl,ls1043a-esdhc", "fsl,esdhc";
 		reg = <0x0 0x1560000 0x0 0x10000>;
 		interrupts = <0 62 0x4>;
 		clock-frequency = <0>;
diff --git a/drivers/mmc/host/sdhci-of-esdhc.c b/drivers/mmc/host/sdhci-of-esdhc.c
index 0ea5903..34e9741 100644
--- a/drivers/mmc/host/sdhci-of-esdhc.c
+++ b/drivers/mmc/host/sdhci-of-esdhc.c
@@ -372,7 +372,8 @@ static void esdhc_get_property(struct platform_device *pdev)
 		host->quirks2 |= SDHCI_QUIRK2_BROKEN_HOST_CONTROL;
 	}
 
-	if (of_device_is_compatible(np, "fsl,ls1021a-esdhc"))
+	if (of_device_is_compatible(np, "fsl,ls1021a-esdhc") ||
+	    of_device_is_compatible(np, "fsl,ls1043a-esdhc"))
 		host->quirks |= SDHCI_QUIRK_BROKEN_TIMEOUT_VAL;
 
 	if (!pltfm_host->clock) {
diff --git a/drivers/mmc/host/sdhci-pltfm.c b/drivers/mmc/host/sdhci-pltfm.c
index 92aa685..d92a483 100644
--- a/drivers/mmc/host/sdhci-pltfm.c
+++ b/drivers/mmc/host/sdhci-pltfm.c
@@ -89,7 +89,8 @@ void sdhci_get_of_property(struct platform_device *pdev)
 		if (of_get_property(np, "no-1-8-v", NULL))
 			host->quirks2 |= SDHCI_QUIRK2_NO_1_8_V;
 
-		if (of_device_is_compatible(np, "fsl,ls2085a-esdhc"))
+		if (of_device_is_compatible(np, "fsl,ls2085a-esdhc") ||
+		    of_device_is_compatible(np, "fsl,ls1043a-esdhc"))
 			host->quirks &= ~SDHCI_QUIRK_BROKEN_CARD_DETECTION;
 
 		if (of_device_is_compatible(np, "fsl,p2020-rev1-esdhc"))
-- 
1.7.5.4

